operator/src/test/java/org/keycloak/operator/PodTemplateE2EIT.java

```diff
@@ -0,0 +1,110 @@
+        Log.info(((operatorDeployment == ClusterOperatorTest.OperatorDeployment.remote) ? "Remote " : "Local ") + "Run Test :" + namespace);
+        // Arrange
+        k8sclient
+                .load(getClass().getResourceAsStream("/example-postgres.yaml"))
+                .inNamespace(namespace)
+                .createOrReplace();
+
+        // Act
+        k8sclient
+                .load(getClass().getResourceAsStream("/correct-podtemplate-keycloak.yml"))
+                .inNamespace(namespace)
+                .createOrReplace();
+
+        // Assert
+        Awaitility
+                .await()
+                .ignoreExceptions()
+                .atMost(3, MINUTES).untilAsserted(() -> {
+            Log.info("Getting logs from Keycloak");
+
+            var keycloakPod = k8sclient
+                    .pods()
+                    .inNamespace(namespace)
+                    .withLabel("app", "keycloak")
+                    .list()
+                    .getItems()
+                    .get(0);
+
+            var logs = k8sclient
+                    .pods()
+                    .inNamespace(namespace)
+                    .withName(keycloakPod.getMetadata().getName())
+                    .getLog();
+
+            Log.info("Full logs are:\n" + logs);
+            assertThat(logs).contains("Hello World");
+            assertThat(keycloakPod.getMetadata().getLabels().get("foo")).isEqualTo("bar");
+        });
+    }
+
+    @Test
+    public void testPodTemplateIncorrect() {
+        Log.info(((operatorDeployment == ClusterOperatorTest.OperatorDeployment.remote) ? "Remote " : "Local ") + "Run Test :" + namespace);
+        // Arrange
+        k8sclient
+                .load(getClass().getResourceAsStream("/example-postgres.yaml"))
+                .inNamespace(namespace)
+                .createOrReplace();
+
+        // Act
+        k8sclient
+                .load(getClass().getResourceAsStream("/incorrect-podtemplate-keycloak.yml"))
+                .inNamespace(namespace)
+                .createOrReplace();
+
+        // Assert
+        Log.info("Getting status of Keycloak");
+        Awaitility
+                .await()
+                .ignoreExceptions()
+                .atMost(3, MINUTES).untilAsserted(() -> {
+                    var message = k8sclient
+                            .resources(Keycloak.class)
+                            .inNamespace(namespace)
+                            .withName("example-podtemplate-kc2")
+                            .get()
+                            .getStatus()
+                            .getConditions()
+                            .stream()
+                            .filter(c -> c.getType().equals(HAS_ERRORS))
+                            .findFirst()
+                            .get()
+                            .getMessage();
+
+                    Log.info("Resource error message is:" + message);
+                    assertThat(message).contains("cannot be modified");
```

Comment:
We should explicitly test all fields that can't be modified.

Reply:
ok, will do 👍

Reply:
done